quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/integration/resteasy/QuarkusHttpRequest.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.quarkus.runtime.integration.resteasy;
+
+import java.io.IOException;
+import java.security.cert.X509Certificate;
+import java.util.Deque;
+import java.util.Iterator;
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.spi.CDI;
+import javax.net.ssl.SSLPeerUnverifiedException;
+import javax.net.ssl.SSLSession;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.UriInfo;
+import org.jboss.resteasy.reactive.common.util.QuarkusMultivaluedHashMap;
+import org.jboss.resteasy.reactive.server.core.ResteasyReactiveRequestContext;
+import org.jboss.resteasy.reactive.server.core.multipart.FormData;
+import org.jboss.resteasy.reactive.server.core.multipart.FormData.FormValue;
+import org.keycloak.http.FormPartValue;
+import org.keycloak.http.HttpRequest;
+import org.keycloak.quarkus.runtime.integration.jaxrs.EmptyMultivaluedMap;
+import org.keycloak.services.FormPartValueImpl;
+
+import io.vertx.ext.web.RoutingContext;
+
+public final class QuarkusHttpRequest implements HttpRequest {
+
+    private static final MultivaluedMap<String, String> EMPTY_FORM_PARAM = new EmptyMultivaluedMap<>();
```

Comment:
Do we really need to create our implementation of empty MultivaluedMap? What's the intention behind this?

What about use `javax.ws.rs.core.MultivaluedHashMap`?
```suggestion
    private static final MultivaluedMap<String, String> EMPTY_FORM_PARAM = new MultivaluedHashMap<>();
```

Reply:
Because we don't want callers writing to it as it should hold no data. If you try to add something to it you get an exception.